src/services/transfer/transfer-out-ehr-core.js

```diff
@@ -18,65 +18,86 @@ export async function transferOutEhrCore({ conversationId, nhsNumber, messageId,
 
-    if (fragmentMessageIds?.length > 0) {
-      await createNewMessageIdsForAllFragments(fragmentMessageIds);
-      logInfo(`Successfully created new message id for all fragments`);
-      ehrCoreWithUpdatedMessageId = await updateReferencedFragmentIds(ehrCoreWithUpdatedMessageId);
-      logInfo(`Successfully replaced fragment id reference in ehrCore`);
-    }
+    const { ehrCoreWithUpdatedMessageId, newMessageId } = await getEhrCoreAndUpdateMessageIds(
+      nhsNumber,
+      conversationId
+    );
+    // TODO: Here we should update the message id of this Registration Request in database to newMessageId.
+    // To be addressed in another ticket
 
+    logInfo('EHR transfer out started');
     logInfo('Sending EHR core');
-    await sendCore(conversationId, odsCode, ehrCoreWithUpdatedMessageId, ehrRequestId, newMessageId);
 
-    await updateConversationStatus(conversationId, Status.SENT_EHR, logs);
-    return defaultResult;
-  } catch (err) {
-    if (err instanceof EhrUrlNotFoundError) {
+    await sendCore(
+      conversationId,
+      odsCode,
+      ehrCoreWithUpdatedMessageId,
+      ehrRequestId,
+      newMessageId
+    );
+
+    await updateConversationStatus(
+      conversationId,
+      Status.SENT_EHR,
+      'EHR has been successfully sent'
+    );
+  } catch (error) {
+    if (error instanceof EhrUrlNotFoundError) {
       await updateConversationStatus(conversationId, Status.MISSING_FROM_REPO);
-      return defaultResult;
     }
-    if (err instanceof DownloadError) {
+    if (error instanceof DownloadError) {
       await updateConversationStatus(conversationId, Status.EHR_DOWNLOAD_FAILED);
-      return defaultResult;
     }
-    logError('EHR transfer out request failed', err);
-    return {
-      hasFailed: true,
-      error: err.message
-    };
+    logError('EHR transfer out request failed', error);
   }
 }
+
+const getEhrCoreAndUpdateMessageIds = async (nhsNumber, conversationId) => {
+  const { ehrCore, fragmentMessageIds } = await getEhrCoreAndFragmentIdsFromRepo(
+    nhsNumber,
+    conversationId
+  );
+
+  let { ehrCoreWithUpdatedMessageId, newMessageId } = await updateMessageIdForEhrCore(ehrCore);
+  logInfo(`Replaced message id for ehrCore`);
+
+  if (fragmentMessageIds?.length > 0) {
+    await createNewMessageIdsForAllFragments(fragmentMessageIds);
+    logInfo(`Created new message id for all fragments`);
+    ehrCoreWithUpdatedMessageId = await updateReferencedFragmentIds(ehrCoreWithUpdatedMessageId);
+    logInfo(`Replaced fragment id references in ehrCore`);
+  }
+  return { ehrCoreWithUpdatedMessageId, newMessageId };
+}
+
+const isEhrRequestDuplicate = async (conversationId) => {
+  const previousTransferOut  = await getRegistrationRequestStatusByConversationId(conversationId);
+  if (previousTransferOut !== null) {
+    logInfo('Duplicate transfer out request');
+    logInfo('EHR out transfer with this conversation ID is already in progress');

```

Comment:
I recon these logs both say the same thing, I would remove one of them and keep the other :)

```suggestion
    logInfo(`EHR out transfer with conversation ID ${conversationId} is already in progress`);
```